config-overrides.js

```diff
@@ -1,18 +1,37 @@
+const CspHtmlWebpackPlugin = require('csp-html-webpack-plugin')
 const _ = require('lodash/fp')
 const rewireReactHotLoader = require('react-app-rewire-hot-loader')
-const { addBabelPlugin } = require('customize-cra')
+const { addBabelPlugin, addWebpackPlugin } = require('customize-cra')
 
 
 module.exports = {
   webpack(config, env) {
-    const newConfig = addBabelPlugin([
-      'prismjs', {
-        languages: ['bash', 'python'],
-        plugins: ['line-numbers'],
-        theme: 'default',
-        css: true
-      }
-    ])(config)
+    const newConfig = _.flow(
+      addBabelPlugin([
+        'prismjs', {
+          languages: ['bash', 'python'],
+          plugins: ['line-numbers'],
+          theme: 'default',
+          css: true
+        }
+      ]),
+      addWebpackPlugin(new CspHtmlWebpackPlugin({
+        // Per our security policy, 'script-src' cannot contain any of the options containing 'unsafe', e.g. 'unsafe-inline'
+        'script-src': [
+          '\'self\'',
```

Comment:
I don't see any difference when moving this to an external script. Advice on how to test?

Reply:
Try to open the app in an unsupported browser, or change the list so yours is unsupported, and see if it warns you.

Reply:
This is it, right?
![Screen Shot 2021-12-16 at 10 22 27](https://user-images.githubusercontent.com/1545444/146399006-efed1a5a-dd21-4876-b70c-aedd9458ea30.png)

Reply:
> Also, do we need `apis.google.com`?

How would we be able to determine this?

Reply:
I would think auth would fail. But I am surprised it isn't failing.
Have we been able to determine definitively whether the CSP is working?
Perhaps trying something like:
1. Pulling in some JS from a trusted site - adding it to our HTML file (do **_not_** add it to the CSP)
2. Commit the code so it is on the PR site where we can test it

We should see an error in the console saying that code violates the CSP - something like:
![image](https://user-images.githubusercontent.com/50371603/146595865-7e13e36f-0185-431e-8612-d802021db71e.png)

Reply:
apis.google.com appears to have a nonce, so I think that one is okay.

Reply:
This specifically takes care of inline scripts for us, too: https://github.com/slackhq/csp-html-webpack-plugin#recommended-configuration

Reply:
Or more accurately, it takes care of any script tags that are present when the page passes through webpack, whether they're inline or links, I think.